widgetssdk/src/main/java/com/glia/widgets/di/UseCaseFactory.java

```diff
@@ -370,4 +372,12 @@ public UnsubscribeFromQueueingStateChangeUseCase createUnSubscribeToQueueingStat
     public SiteInfoUseCase createSiteInfoUseCase() {
         return new SiteInfoUseCase(repositoryFactory.getGliaEngagementRepository());
     }
+
+    public GliaSurveyUseCase getGliaSurveyUseCase() {
+        return GliaSurveyUseCase.getInstance(createOnEngagementUseCase(), createOnEngagementEndUseCase());
```

Comment:
You could hold singleton here - usecase shouldnt hold its own instance beside main responsibility